### PR TITLE
Visual studio warnings

### PR DESCRIPTION
**Pull request by jvalenzuela**
Various goodies in our quest to keep compilers happy.

### CODE REVIEW

**Comment by jvalenzuela** [comment]
No problem. I considered /Za option, or more specifically, I'm compiling with /W4; I thought it better to stick with standard(ANSI) code, especially since non-standard extensions aren't needed for struct initialization. Same applies to GCC; it also has it's own extensions, but I don't use them unless there is no other option.

File: source/src/enet_encap/encap.c [hunk 1]
```diff
@@ -781,8 +781,14 @@ EipInt16 CreateEncapsulationStructure(const EipUint8 *receive_buffer,
...
   encapsulation_data->current_communication_buffer_position =
     (EipUint8 *) receive_buffer;
-  return (receive_buffer_length - ENCAPSULATION_HEADER_LENGTH
-          - encapsulation_data->data_length);
+
+  const size_t diff = receive_buffer_length - ENCAPSULATION_HEADER_LENGTH
+          - encapsulation_data->data_length;
+
+  /* Sanity check before casting the return value to EipInt16. */
+  OPENER_ASSERT((INT16_MIN <= diff) && (INT16_MAX >= diff));
```
Comment (CapXilinx) [inline source/src/enet_encap/encap.c:803]: This assert is always false in my test case

Comment (CapXilinx) [reply source/src/enet_encap/encap.c:803]: Also, diff as a size_t can never be a negative value

Comment (jvalenzuela) [reply source/src/enet_encap/encap.c]: Seems I needed to brush up on C integer conversions. It should be fixed in 628cb63.

**Comment by jvalenzuela** [comment]
These changes are being updated to work with the current master branch.

**Pull request closed without merging**